php-checks/src/main/java/org/sonar/php/checks/utils/FunctionArgumentCheck.java

```diff
@@ -0,0 +1,181 @@
+import org.sonar.plugins.php.api.tree.expression.LiteralTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+/**
+ * This abstract class simplifies the checking of function calls for the specific arguments.
+ * {@link ArgumentVerifier} can be used to check whether an argument corresponds to a certain value or a set of values.
+ * Based on a flag of the verifier an issue can be created on match or non-match.
+ * If arguments should only be checked when other arguments must match certain values,
+ * {@link ArgumentIndicator} can be used. They are used as a condition for the verification.
+ */
+public abstract class FunctionArgumentCheck extends PHPVisitorCheck {
+
+  private AssignmentExpressionVisitor assignmentExpressionVisitor;
+
+  /**
+   * Implement this method to create an issue with specific message for the certain check.
+   * As tree argument the verified argument is given.
+   */
+  protected abstract void createIssue(ExpressionTree tree);
+
+  /**
+   * Several {@link ArgumentVerifier} and {@link ArgumentIndicator} can be included in the check.
+   * This means a single function can be checked for multiple arguments.
+   *
+   * The order of indicators and verifiers must be observed.
+   * On the one hand, an indicator can lead to the termination of the check of a certain function.
+   * On the other hand, issues can be created by verifier before an indicator is triggered.
+   */
+  public void checkArgument(FunctionCallTree tree, String expectedFunctionName, ArgumentIndicator... expectedArgument) {
+    String functionName = CheckUtils.getLowerCaseFunctionName(tree);
+    List<ExpressionTree> arguments = tree.arguments();
+
+    if (expectedFunctionName.equals(functionName)) {
+      for (ArgumentIndicator argumentIndicator: expectedArgument) {
+        if (argumentIndicator.position >= arguments.size() || !verifyArgument(arguments, argumentIndicator)) {
+          return;
+        }
+      }
+    }
+  }
+
+  @Override
+  public void visitCompilationUnit(CompilationUnitTree tree) {
+    assignmentExpressionVisitor = new AssignmentExpressionVisitor(context().symbolTable());
+    tree.accept(assignmentExpressionVisitor);
+    super.visitCompilationUnit(tree);
+  }
+
+  private boolean verifyArgument(List<ExpressionTree> arguments, ArgumentIndicator argumentIndicator) {
+    ExpressionTree argument = arguments.get(argumentIndicator.position);
+    ExpressionTree argumentValue = getAssignedValue(argument);
+
+    Optional<String> value = nameOf(argumentValue);
+    if (value.isPresent()) {
+      String quoteLessLowercaseValue = CheckUtils.trimQuotes(value.get()).toLowerCase(Locale.ENGLISH);
+      boolean containValues = argumentIndicator.values.contains(quoteLessLowercaseValue);
+
+      if (argumentIndicator instanceof ArgumentVerifier && ((ArgumentVerifier) argumentIndicator).raiseIssueOnMatch == containValues) {
+        createIssue(argument);
+        return true;
```

Comment:
What does the return value mean for `verifyArgument`? If I change this line to `return false`, unit tests still pass.

Reply:
If multiple ArgumentVerifiers are used they can be found in a function call. A bool returned must be used, because the loop must be interrupted by an identifier.